dataprofiler/tests/profilers/test_unstructured_text_profile.py

```diff
@@ -164,4 +165,95 @@ def test_case_sensitivity(self):
             expected_word_count = {'hello': 1, 'name': 1, 'grant': 2, 'bob': 1,
                                    'friends': 1}
             self.assertDictEqual(expected_word_count, profile['word_count'])
-        
\ No newline at end of file
+
+    def test_text_profile_with_wrong_options(self):
+        with self.assertRaisesRegex(ValueError,
+                "TextProfiler parameter 'options' must be of type"
+                " TextProfilerOptions."):
+            TextProfiler("Name", options="wrong_data_type")
+
+    def test_options_default(self):
+        options = TextProfilerOptions()
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is test, a Test sentence.!!!"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'sentence': 1, 'Test': 1, 'test': 1}
+        expected_vocab = [' ', ',', '.', '!', 'T', 'h', 'i', 's', 'a',
+                          'e', 't', 'n', 'c']
+        self.assertDictEqual(expected_word_count, profile['word_count'])
+        self.assertCountEqual(expected_vocab, profile['vocab'])
+
+    def test_options_case_sensitive(self):
+        options = TextProfilerOptions()
+        options.is_case_sensitive = False
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is test, a Test sentence.!!!"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'sentence': 1, 'test': 2}
+        expected_vocab = [' ', ',', '.', '!', 'T', 'h', 'i', 's', 'a',
+                          'e', 't', 'n', 'c']
+        self.assertDictEqual(expected_word_count, profile['word_count'])
+        self.assertCountEqual(expected_vocab, profile['vocab'])
+
+    def test_options_stop_words(self):
+        # with a list of stopwords
+        options = TextProfilerOptions()
+        options.stop_words = ['hello', 'sentence', 'is', 'a']
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is test, a Test sentence.!!!"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'This': 1, 'Test': 1, 'test': 1}
+        expected_vocab = [' ', ',', '.', '!', 'T', 'h', 'i', 's', 'a',
+                          'e', 't', 'n', 'c']
```

Comment:
What happened to the comma, period, and explanation point here, because they were not in the provided list of stop words, but also weren't present in the word_count dictionary?

Reply:
also, it looks like `is` and `a` were included in the vocab. Group question, should `stop` words prevent vocab from being read?

Could be decided in subsequent PR.

Reply:
usually, word-processing doesn't care about punctuations. So, they are all excluded from word update.
About the consistency of word and vocab updates on the character, we can discuss further, but may be in the next PR.